bears/general/LineCountBear.py

```diff
@@ -18,24 +19,58 @@ def _get_blank_line_count(self, file):
         return num_blank_lines
 
     def run(self, filename, file, max_lines_per_file: int,
-            exclude_blank_lines: bool=False):
+            ignore_files_regex: typed_list(str),
+            exclude_blank_lines: bool=False,
+            min_lines_per_file: int=1):
         """
-        Count the number of lines in a file and ensure that they are
-        smaller than a given size.
+        Count the number of lines in a file and ensure that they lie within
+        the range of given sizes. There are files for which lower limit isn't
+        required, but these files still shouldn't be rejected. Some of such
+        files when minimum is 1 are mentioned in ignore_files_regex
+        in the repository .coafile.
 
-        :param max_lines_per_file: Number of lines allowed per file.
-        :param exclude_blank_lines: "True" if blank lines are to be excluded.
+        :param max_lines_per_file:
+            Maximum number of lines allowed per file.
+        :param exclude_blank_lines:
+            ``True`` if blank lines are to be excluded.
+        :param min_lines_per_file:
+            Minimum number of lines required per file.
+        :param ignore_files_regex:
+            Files, for which minimum is not required.
         """
         file_length = len(file)
+        ignore_regexes = [re.compile(regex)
+                          for regex in ignore_files_regex]
+
+        if min_lines_per_file > max_lines_per_file:
+            yield Result.from_values(
+                origin=self,
+                message=('Allowed maximum lines per file({}) is smaller than '
+                         'minimum lines per file({}) and this is prohibited'
+                         .format(max_lines_per_file,
+                                 min_lines_per_file)),
+                severity=RESULT_SEVERITY.NORMAL,
+                file=filename)
+            return
         if exclude_blank_lines:
             num_blank_lines = self._get_blank_line_count(file)
             file_length = file_length - num_blank_lines
         if file_length > max_lines_per_file:
             yield Result.from_values(
                 origin=self,
-                message=('This file had {count} lines, which is {extra} '
+                message=('This file had {} lines, which is {} '
                          'lines more than the maximum limit specified.'
-                         .format(count=file_length,
-                                 extra=file_length-max_lines_per_file)),
+                         .format(file_length,
+                                 file_length-max_lines_per_file)),
```

Comment:
Definitely unnecessary change. Please file an issue to discuss removing named placeholders if the format-string is simple enough.

Reply:
@Makman2 This have been suggested by previous reviewer :p

Reply:
Not part of your desired change :+1: Could be a good newcomer issue anyway :+1: